File: customer-service/src/main/java/com/wks/servicemarketplace/customerservice/core/usecase/address/CountryCode.java
package com.wks.servicemarketplace.customerservice.core.usecase.address;

import com.wks.servicemarketplace.customerservice.core.exceptions.CountryCodeNotFoundException;
import lombok.NonNull;
import lombok.Value;

import java.util.Optional;

@Value
public class CountryCode {

    @NonNull
    private final String countryCode;

    public CountryCode(String countryCode) throws CountryCodeNotFoundException {
        this.countryCode = Optional.ofNullable(com.neovisionaries.i18n.CountryCode.getByAlpha2Code(countryCode))
                .map(com.neovisionaries.i18n.CountryCode::name)
                .orElseThrow(() -> new CountryCodeNotFoundException(countryCode, "ISO 3166-1 alpha-2"));
    }
}

File: order-service/src/main/kotlin/com/wks/servicesmarketplace/orderservice/core/CountryCode.kt
package com.wks.servicesmarketplace.orderservice.core

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.wks.servicesmarketplace.orderservice.core.exceptions.CountryNotFoundException
import javax.persistence.Embeddable

@Embeddable
class CountryCode internal constructor(@JsonValue var value: String) {
    companion object {
        @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
        @JvmStatic
        fun of(countryCode: String): CountryCode {
            com.neovisionaries.i18n.CountryCode.getByAlpha2Code(countryCode)
                    ?: throw CountryNotFoundException(countryCode, "ISO 3166-1 alpha-2")
            return CountryCode(countryCode)
        }
    }

    override fun toString() = value
}